bundles/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/util/ParadoxUtil.java

```diff
@@ -54,27 +56,54 @@ public static void printPacket(String description, byte[] array) {
         }
     }
 
-    private static void printByteArray(String description, byte[] array, int length) {
-        if (description != null && !description.isEmpty()) {
-            logger.trace("{}", description);
+    public static void printByteArray(String description, byte[] array) {
+        if (array == null) {
+            logger.trace("Array is null");
+            return;
         }
-        int countBytes = 0;
-        String result = "";
-        for (int index = 0; index < length; index++) {
-            countBytes++;
-            String st = String.format("0x%02X,\t", array[index]);
-            result += st;
-            if (countBytes > 7) {
-                logger.trace(result);
-                countBytes = 0;
-                result = "";
-                continue;
-            }
+        printByteArray(description, array, array.length);
+    }
+
+    public static void printByteArray(String description, byte[] array, int length) {
+        if (!logger.isTraceEnabled()) {
+            return;
         }
+
+        String result = byteArrayToString(array, length);
         if (!result.isEmpty()) {
-            logger.trace(result);
+            logger.trace("{}", description + SPACE_DELIMITER + result);
         }
+    }
 
+    public static String byteArrayToString(byte[] array) {
+        return byteArrayToString(array, array.length);
+    }
+
+    /**
+     *
+     * Returns passed array as HEX string. On every 8 bytes we put space for better readability. Example 16
+     * bytes array output: AA47000263000000 03EE00EEEEEEB727
+     *
+     * @param array
+     * @param length
+     * @return String
+     */
+    public static String byteArrayToString(byte[] array, int length) {

```

Comment:
Why not use `HexUtils.bytestoHex`  from the core?

Reply:
When packet is encrypted, Paradox is adding 0xEE to the payload to every 16, 32, 48, 64, etc bytes. In the header there is a field which defines the actual payload length. I'd like to be able to print only the actual length for debugging purposes in TRACE level. By providing length, I cut the array to the length passed as an argument. Also I split on every 8 bytes by adding a space which is not possible in HexUtils, for easier reading. I guess if I do pre-processing I could use HexUtils without the every 8 bytes space functionality though...